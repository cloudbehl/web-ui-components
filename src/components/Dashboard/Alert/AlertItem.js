import React from 'react';
import PropTypes from 'prop-types';
import { get } from 'lodash';
import { Col, Row, Icon } from 'patternfly-react';

const getSeverityIcon = severity => {
  switch (severity) {
    case 'critical':
      return <Icon type="fa" name="exclamation-circle" className="kubevirt-alert__icon-critical" />;
    case 'warning':
    default:
      return <Icon type="fa" name="exclamation-triangle" className="kubevirt-alert__icon-warning" />;
  }
};

export const AlertItem = ({ alert }) => (
  <Row className="kubevirt-alert__item">
    <Col lg={1} md={1} sm={1} xs={1}>
      {getSeverityIcon(get(alert, 'labels.severity'))}
    </Col>
    <Col lg={11} md={11} sm={11} xs={11}>
      {get(alert, 'annotations.message')}
    </Col>
  </Row>
);

AlertItem.propTypes = {
  alert: PropTypes.object.isRequired,
};
