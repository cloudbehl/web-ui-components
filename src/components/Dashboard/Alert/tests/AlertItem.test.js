import React from 'react';
import { shallow } from 'enzyme';

import { AlertItem } from '../AlertItem';
import { default as AlertItemFixtures } from '../fixtures/AlertItem.fixture';

const testAlertItem = () => <AlertItem {...AlertItemFixtures[0].props} />;

describe('<AlertItem />', () => {
  it('renders correctly', () => {
    const component = shallow(testAlertItem());
    expect(component).toMatchSnapshot();
  });
});
