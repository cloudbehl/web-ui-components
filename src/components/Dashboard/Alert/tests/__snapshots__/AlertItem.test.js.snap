// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertItem /> renders correctly 1`] = `
<div
  className="kubevirt-alert__item"
>
  <Icon
    className="kubevirt-alert__icon kubevirt-alert__icon--warning"
    name="exclamation-triangle"
    type="fa"
  />
  <div
    className="kubevirt-alert__item-message"
  >
    20% of the metrics targets are down.
  </div>
</div>
`;
